### DISCUSSION TITLE
Have styled-components preffer TSTag group

### DISCUSSION THREAD

**Discussion by vuki656** [nvim-treesitter/nvim-treesitter#1247] [Q&A]
Hey, right now styled-components are tagged with the following groups

![image](https://user-images.githubusercontent.com/8104698/115966446-e213ec80-a52d-11eb-99a8-53bad22742ab.png)

Is there any way to make them prefer the `TSTag` group over any other since they "technically" should be considered as `TSTag`s

Basically, I set a `TSTag` color and would like styled-components to use them so they are the same. So NOT this:

![image](https://user-images.githubusercontent.com/8104698/115966499-1be4f300-a52e-11eb-9dc4-f3b588b86aac.png)

I'm aware I can set the color for other groups listed for styled-components but those groups are used for other things that I don't to have the same color for all of them if that makes sense.

**Comment by vigoux** [comment]
Hi, and sorry for the delay.

I am not sure I understand the issue here, but let me rephrase it to be sure: you would like `TSTag` to be on top of the highlight stack that is here, right ?

**Reply by theHamsta** [reply]
There is a special rule to use `Type` for tags with upper case letters for components in the React highlight.scm.
https://github.com/theHamsta/nvim-treesitter/blob/04953f7b233588ea39836599e8da9c3d2a63c3a4/queries/jsx/highlights.scm#L17-L31

The ideal solution for this user would be to overwrite the `jsx/highlight.scm`. We could offer something in the README like
```lua
local jsx_query = table.concat(vim.tbl_filter(function(f) not f:find('nvim-treesitter', 1, true) end, require("vim.treesitter.query").get_query_files(jsx, "highlight", true)), '\n')
require("vim.treesitter.query").set_query("jsx", "highlight", jsx_query)
``` 
on how to skip a specific file from nvim-treesitter. Probably `set_query` should also accept a function that does the resolving only at the moment the query is needed

**Reply by theHamsta** [reply]
Or they can just copy the referenced lines in their own `after/queries/jsx/highlights.scm` with `@tag` instead of `@type`/`@constructor`
